cmd/soci/commands/internal/index.go

```diff
@@ -0,0 +1,40 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package internal
+
+import "github.com/urfave/cli"
+
+const (
+	ExistingIndexFlagName = "existing-index"
+	Warn                  = "warn"
+	Abort                 = "abort"
+	Skip                  = "skip"
+	Allow                 = "allow"
+)
+
+var ValidExistingIndexOptions = map[string]bool{
+	Warn:  true,
+	Abort: true,
+	Skip:  true,
+	Allow: true,
+}
+
+var ExistingIndexFlag = cli.StringFlag{
+	Name:  ExistingIndexFlagName,
+	Value: Warn,
+	Usage: "Configure how to handle existing SOCI artifacts in remote when pushing indices",
```

Comment:
I think think it would be useful to have explicit explanations about what happens with each option and what the default is.